test/build_processed_data_test.py

```diff
@@ -1,6 +1,57 @@
-from libs import build_processed_dataset
+from unittest.mock import patch
 
+import pytest
 
-def test_get_testing_df():
-    testing_df = build_processed_dataset.get_testing_timeseries_by_state("MA")
-    assert testing_df is not None
+from libs import build_processed_dataset, validate_results
+from libs.datasets import CommonFields, CDSDataset, CovidTrackingDataSource
+import pandas as pd
+from libs.datasets.results_schema import CUMULATIVE_POSITIVE_TESTS, CUMULATIVE_NEGATIVE_TESTS
+
+from libs.enums import Intervention
+from more_itertools import one
+
+
+def test_get_usa_by_states_df():
+    empty_df_with_state = pd.DataFrame([], columns=["state",])
+    with patch(
+        "libs.build_processed_dataset.get_state_projections_df", return_value=empty_df_with_state
+    ):
+        df = build_processed_dataset.get_usa_by_states_df(
+            input_dir="/tmp", intervention_type=Intervention.OBSERVED_INTERVENTION.value
+        )
+    validate_results.validate_states_df("TX", df)
+
+    tx_record = one(df.loc[df[CommonFields.STATE_FULL_NAME] == "Texas"].to_dict(orient="records"))
+    assert tx_record[CUMULATIVE_POSITIVE_TESTS] > 100
+    assert tx_record[CUMULATIVE_NEGATIVE_TESTS] > 100
+
+
+def test_get_testing_df_by_state():
+    positive_field = CovidTrackingDataSource.Fields.POSITIVE_TESTS
+    negative_field = CovidTrackingDataSource.Fields.NEGATIVE_TESTS
+    df = build_processed_dataset.get_testing_timeseries_by_state("MA")
+    assert positive_field in df.columns
+    assert negative_field in df.columns
+    # TODO(tom): have build_processed_dataset return a real datetime dtype so callers don't need to parse a string.
+    df[CommonFields.DATE] = pd.to_datetime(df[CommonFields.DATE], format="%m/%d/%y")
+    df.set_index(CommonFields.DATE, inplace=True)
+    # No joke, our only source of state-level timeseries testing data is CDSDataset and it has NaN for MA until April 6.
```

Comment:
Do you mean covid tracking ?

Reply:
Also I wonder if we can just import this from the MA data - I feel like they reported it?

Reply:
The next PR will be tricky... `grep 20200401,MA ../covid-data-public/data/covid-tracking/covid_tracking_states.csv` does find positive and negative values, but the way the data is combined this test fails when it is produced from combined data. Lets get this merged and move on to that next...